plugins/CoreAdminHome/Controller.php

```diff
@@ -154,20 +155,21 @@ public function setMailSettings()
             $this->checkTokenInUrl();
 
             // Update email settings
-            $mail = array();
-            $mail['transport'] = (Common::getRequestVar('mailUseSmtp') == '1') ? 'smtp' : '';
-            $mail['port'] = Common::getRequestVar('mailPort', '');
-            $mail['host'] = Common::unsanitizeInputValue(Common::getRequestVar('mailHost', ''));
-            $mail['type'] = Common::getRequestVar('mailType', '');
-            $mail['username'] = Common::unsanitizeInputValue(Common::getRequestVar('mailUsername', ''));
-            $mail['password'] = Common::unsanitizeInputValue(Common::getRequestVar('mailPassword', ''));
+            $request = Request::fromRequest();
```

Comment:
would it make sense to rather use dependency injection here? That can improve testability. Especially when it's also used in other places. Maybe the class definition would look differently or so.

The only thing is maybe that when $_GET/$_POST changes during the request, and the class is stored in DI, we need to make sure we don't work of a copy of these parameters. There's benefits and downsides to this but thinking if testability it could overweight some downsides under circumstances. Just a thought.

Reply:
I thought about DI in the first place, but discarded it due to the different use cases like building a request from GET and/or POST, a query string or even a custom array, I don't think DI would have such a big benefit here. We could though make it possible to fully replace the class. But if we always only work on one instance it might cause problems mapping the different use cases.
And for testing purpose you can simply overwrite $_GET and $_POST, as the class works with those in most cases.

Reply:
OK that works too. Could have always used `Request` and `PostRequest` or something with different classes if different ones needed. Not too important though I suppose.